mount/upgrader.go

```diff
@@ -66,9 +72,13 @@ func (u *Upgrader) Fetch(ctx context.Context) (Reader, error) {
 	// after it's done, open the resulting transient.
 	u.lk.Lock()
 	if u.transient != "" {
+		log.Debugw("have an existing transient copy, checking if we can use it for the fetch", "shard", u.key, "transient", u.transient)
 		if _, err := os.Stat(u.transient); err == nil {
-			u.lk.Unlock()
+			log.Debugw("transient is still alive, not refetching", "shard", u.key, "transient", u.transient)
+			defer u.lk.Unlock()
```

Comment:
@raulk This was the unlock bug I was talking about. Hairy edge case that I think actually happened on the Sofia Miner.